### PR TITLE
Tasks enhancement

### LINKED ISSUE #828: Tasks enhancement
- Task definition: add the page Task Execution (list)
- Task definition: launch and destroy actions should be move in the header (title)
- Launch: fix the cancel action

### PR DESCRIPTION
**Pull request by oodamien**
- Task definition: add the page Task Execution (list)
- Task definition: launch and destroy actions are moved in the header page
- Launch: fix the cancel action
Resolves #828

### CODE REVIEW

File: ui/src/app/tasks/task-definition/executions/task-definition-executions.component.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    this.params$.complete();
+    this.params$.unsubscribe();
+  }
+
+  /**
+   * Refresh
+   * Create an observable which provides the required data
+   * @param {TaskExecutionListParams} params
+   */
+  refresh(params: TaskExecutionListParams) {
```
Comment (ilayaperumalg) [inline ui/src/app/tasks/task-definition/executions/task-definition-executions.component.ts]: I guess all the paging support can be abstracted away to a common module for reusability.

**Comment by ilayaperumalg** [comment]
After applying the rebased changes, I keep getting the following build failure:

```
[INFO] > spring-cloud-dataflow-ui@1.3.0 mavenbuild-without-tests /Users/igopinathan/dev/git/spring-cloud/spring-cloud-dataflow-ui/ui
[INFO] > node ./node_modules/@angular/cli/bin/ng build --prod --output-path ./../target/classes/public/dashboard --base-href /dashboard/
[INFO]
[INFO] Date: 2018-06-28T18:58:22.706Z
[INFO] Hash: f48990aba74cebc00009
[INFO] Time: 10959ms
[INFO] chunk {0} styles.b018edcd321b61ea40a9.bundle.css (styles) 244 kB [initial] [rendered]
[INFO] chunk {1} polyfills.997d8cc03812de50ae67.bundle.js (polyfills) 84 bytes [initial] [rendered]
[INFO] chunk {2} main.ee32620ecd1edff94184.bundle.js (main) 84 bytes [initial] [rendered]
[INFO] chunk {3} inline.318b50c57b4eba3d437b.bundle.js (inline) 796 bytes [entry] [rendered]
[ERROR]
[ERROR] ERROR in src/app/shared/flo/support/shape-component.ts(2,21): error TS2307: Cannot find module 'jointjs'.
[ERROR] src/app/shared/flo/decoration/decoration.component.ts(2,21): error TS2307: Cannot find module 'jointjs'.
[ERROR] node_modules/spring-flo/dist/src/shared/flo-common.d.ts(1,24): error TS2307: Cannot find module 'jointjs'.
[ERROR] node_modules/spring-flo/dist/src/palette/palette.component.d.ts(2,21): error TS2307: Cannot find module 'jointjs'.
[ERROR] node_modules/spring-flo/dist/src/editor/editor.component.d.ts(2,21): error TS2307: Cannot find module 'jointjs'.
```

The travis failed on some other reason. @oodamien were you able to successfully build locally?

File: ui/src/app/tasks/task-definition/executions/task-definition-executions.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    });
+
+    it('should navigate to the detail page', () => {
+      const line: DebugElement = fixture.debugElement.queryAll(By.css('#taskExecutionsTable tbody tr'))[0];
+      const navigate = spyOn((<any>component).router, 'navigate');
+      line.query(By.css('.actions button[name="task-details0"]')).nativeElement.click();
+      expect(navigate).toHaveBeenCalledWith(['tasks/executions/2']);
+    });
+
+    it('should navigate to the detail page', () => {
```
Comment (BoykoAlex) [inline ui/src/app/tasks/task-definition/executions/task-definition-executions.component.spec.ts]: This test always fails for me. Navigate never called...

File: ui/src/app/tasks/components/tasks.interface.ts [hunk 3]
```diff
@@ -18,3 +18,11 @@ export interface TaskCreateParams {
   definition: string;
   name: string;
 }
+
+export interface TaskExecutionListParams {
+  task: string;
+  page: number;
```
[the remark this answers is not in the document]
Comment (oodamien) [reply ui/src/app/tasks/components/tasks.interface.ts]: Should extend ListParams which the reusable paging

**Comment by BoykoAlex** [comment]
Merged with 6b94094190178388508f5ece00bb4b951f926100

**Pull request closed without merging**